birdset/modules/models/beats.py

```diff
@@ -26,18 +26,18 @@ def __init__(
         # self.classifier = nn.Linear(
         #     in_features=self.EMBEDDING_SIZE, out_features=num_classes
         # )
-        self.classifier = nn.Sequential(
-            nn.Linear(self.EMBEDDING_SIZE, 128),
-            nn.ReLU(),
-            nn.Dropout(0.5),
-            nn.Linear(128, 64),
-            nn.ReLU(),
-            nn.Linear(64, self.num_classes),
-        )
-        
-        # freeze the model
-        #! Dont freeze when finetuning
         if self.train_classifier:
+            self.classifier = nn.Sequential(
+                nn.Linear(self.EMBEDDING_SIZE, 128),
+                nn.ReLU(),
+                nn.Dropout(0.5),
+                nn.Linear(128, 64),
+                nn.ReLU(),
+                nn.Linear(64, self.num_classes),
+            )
+        
+            # freeze the model
+            #! Dont freeze when finetuning
             for param in self.model.parameters():
                 param.requires_grad = False
 
```

Comment:
To make sure all weights are trained in fine-tuning we need to make sure that the models don't have a frozen backbone. Because the fine-tuning module doesn't need the train_classifier (param=False) I moved freezing the backbone into the if statement for now. This was already the case for some models.

Reply:
The default of these models was that the backbone is always frozen so this shouldn't break other experiments as most need the classifier head